src/accelerate/accelerator.py

```diff
@@ -280,16 +281,23 @@ def __init__(
 
         dynamo_plugin = TorchDynamoPlugin() if dynamo_backend is None else TorchDynamoPlugin(backend=dynamo_backend)
 
-        if deepspeed_plugin is None:  # init from env variables
-            deepspeed_plugin = (
-                DeepSpeedPlugin() if os.environ.get("ACCELERATE_USE_DEEPSPEED", "false") == "true" else None
-            )
+        if deepspeed_plugin is None:
+            # First check if we're creating another `Accelerator` w/o setting `deepspeed_plugin`
+            if PartialState().distributed_type == DistributedType.DEEPSPEED:
+                deepspeed_plugin = AcceleratorState().deepspeed_plugins
+            else:
+                # init from env variables
+                deepspeed_plugin = (
+                    DeepSpeedPlugin() if os.environ.get("ACCELERATE_USE_DEEPSPEED", "false") == "true" else None
+                )
         else:
-            assert isinstance(
-                deepspeed_plugin, DeepSpeedPlugin
-            ), "`deepspeed_plugin` must be an `accelerate.utils.DeepSpeedPlugin` object."
+            if isinstance(deepspeed_plugin, dict):
+                for plugin in deepspeed_plugin.values():
+                    if not isinstance(plugin, DeepSpeedPlugin):
+                        raise TypeError("`deepspeed_plugin` must be a DeepSpeedPlugin object.")
+
+        if deepspeed_plugin is not None:
```

Comment:
maybe raise a warning if the user pass another deepspeed plugin in the second accelerator ?